pkg/descheduler/pod/pods.go

```diff
@@ -68,3 +69,27 @@ func IsBurstablePod(pod *v1.Pod) bool {
 func IsGuaranteedPod(pod *v1.Pod) bool {
 	return utils.GetPodQOS(pod) == v1.PodQOSGuaranteed
 }
+
+// SortPodsBasedOnPriorityLowToHigh sorts pods based on their priorities from low to high.
+// If pods have same priorities, they will be sorted by QoS in the following order:
+// BestEffort, Burstable, Guaranteed
+func SortPodsBasedOnPriorityLowToHigh(pods []*v1.Pod) {
```

Comment:
What about to have:
```
func PodPrioritySortLowToHigh(i, j int) bool {
	if pods[i].Spec.Priority == nil && pods[j].Spec.Priority != nil {
		return true
	}
	if pods[j].Spec.Priority == nil && pods[i].Spec.Priority != nil {
		return false
	}
	if (pods[j].Spec.Priority == nil && pods[i].Spec.Priority == nil) || (*pods[i].Spec.Priority == *pods[j].Spec.Priority) {
		if IsBestEffortPod(pods[i]) {
			return true
		}
		if IsBurstablePod(pods[i]) && IsGuaranteedPod(pods[j]) {
			return true
		}
		return false
	}
	return *pods[i].Spec.Priority < *pods[j].Spec.Priority
}
```
instead and replace all invocations of `SortPodsBasedOnPriorityLowToHigh` with `sort.Slice(pods, PodPrioritySortLowToHigh`? So the implementation is minimal.

Reply:
This implementation looks like [less func](https://github.com/kubernetes/kubernetes/blob/e7ca64fbe16d0c4b6c7b36aecde9cd75042b2828/pkg/scheduler/framework/v1alpha1/interface.go#L214) of QueueSortPlugin, as  func in `Slice(slice interface{}, less func(i, j int)` only receives two int params, we may implement and use it like:

```
func PodComparePriorityLowToHigh(i, j pod) bool {...}

sort.Slice(pods, func(i, j int) bool {
        return PodComparePriorityLowToHigh(pod[i], pod[j])
}
```

That looks a little complex, besides I think `pod util` should provide a `sort func` and not only a `less func`, maybe we can provide both and refactor `sort func` like:
```
func PodComparePriorityLowToHigh(i, j pod) bool {...}

func SortPodsBasedOnPriorityLowToHigh(pods []*v1.Pod) {
        sort.Slice(pods, func(i, j int) bool {
                return PodComparePriorityLowToHigh(pod[i], pod[j])
        }
}
```
What do you think @ingvagabund ?

Reply:
Sorry about that. Yeah, you are right. Too much lambda calculus in my head :) We can elaborate on that later once we have more callers of the same "less" implementation for sorting.